server/webhook.go

```diff
@@ -170,146 +100,170 @@ func (s *Server) registerWebhookRoutes(g *echo.Group) {
-							Message:     commit.Message,
-							CreatedTs:   commit.Timestamp.Unix(),
-							URL:         commit.URL,
-							AuthorName:  commit.Author.Name,
-							AuthorEmail: commit.Author.Email,
-							Added:       common.EscapeForLogging(file.name),
-						},
-					}
-					createdMessage, created, activityCreateList, httpErr := s.createIssueFromPushEvent(
-						ctx,
-						pushEvent,
-						repo,
-						webhookEndpointID,
-						file.name,
-						file.itemType,
-					)
-					if httpErr != nil {
-						return httpErr
-					}
-					if created {
-						createdMessageList = append(createdMessageList, createdMessage)
-					}
-					repoID2ActivityCreateList[repo.ID] = append(repoID2ActivityCreateList[repo.ID], activityCreateList...)
-				}
-				if len(createdMessageList) == 0 {
-					log.Debug("Ignored push event file because no applicable file found in the commit list", zap.String("fileName", file.name), zap.Any("repos", handleRepos))
-					for _, repo := range handleRepos {
-						if activityCreateList, ok := repoID2ActivityCreateList[repo.ID]; ok {
-							for _, activityCreate := range activityCreateList {
-								if _, err = s.ActivityManager.CreateActivity(ctx, activityCreate, &ActivityMeta{}); err != nil {
-									log.Warn("Failed to create project activity for the ignored repository file",
-										zap.Error(err),
-									)
-								}
-							}
+			repoID2ActivityCreateList[repo.ID] = append(repoID2ActivityCreateList[repo.ID], activityCreateList...)
+		}
+		if len(createdMessageList) == 0 {
+			for _, repo := range filteredRepos {
+				if activityCreateList, ok := repoID2ActivityCreateList[repo.ID]; ok {
+					for _, activityCreate := range activityCreateList {
+						if _, err := s.ActivityManager.CreateActivity(ctx, activityCreate, &ActivityMeta{}); err != nil {
+							log.Warn("Failed to create project activity for the ignored repository file",
+								zap.Error(err),
+							)
 						}
 					}
 				}
-				createdMessages = append(createdMessages, createdMessageList...)
 			}
 		}
+		createdMessages = append(createdMessages, createdMessageList...)
+	}
+	if len(createdMessages) == 0 {
+		log.Warn("Ignored push event because no applicable file found in the commit list", zap.Any("repos", filteredRepos))
+	}
+	return createdMessages, nil
+}
+
+func (s *Server) validateWebhookRequest(ctx context.Context, c echo.Context) (string, []byte, []*api.Repository, *echo.HTTPError) {
+	webhookEndpointID := c.Param("id")
+	repos, err := s.store.FindRepository(ctx, &api.RepositoryFind{WebhookEndpointID: &webhookEndpointID})
+	if err != nil {
+		return "", nil, nil, echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to respond webhook event for endpoint: %v", webhookEndpointID)).SetInternal(err)
+	}
+	if len(repos) == 0 {
+		return "", nil, nil, echo.NewHTTPError(http.StatusNotFound, fmt.Sprintf("Webhook endpoint not found: %v", webhookEndpointID))
+	}
+	body, err := io.ReadAll(c.Request().Body)
```

Comment:
Reading body doesn't seem to belong to validateWebhookRequest(). This function doesn't seem to be doing validation. Let's either move the content back to caller or rename the function to its actual usage.